simple-facebook-instant-articles.php

```diff
@@ -425,6 +427,35 @@ public function render_images( \DOMDocument &$dom, \DOMXPath &$xpath ) {
 	}
 
 	/**
+	 * Facebook throws a warning for all headings below h2.
+	 *
+	 * Replace with h2s.
+	 *
+	 * @param  \DOMDocument &$dom   DOM object generated for post content.
+	 * @param  \DOMXPath    &$xpath XPATH object generated for post content.
+	 *
+	 * @return void
+	 */
+	public function fix_headings( \DOMDocument &$dom, \DOMXPath &$xpath ) {
+
+		$headings = array( 'h3', 'h4', 'h5', 'h6' );
+
+		foreach ( $headings as $heading_tag ) {
+
+			foreach ( $dom->getElementsByTagName( $heading_tag ) as $node ) {
+
+				$h2 = $dom->createElement( 'h2' );
+
+				while ( $node->childNodes->length > 0 ) {
+					$h2->appendChild( $node->childNodes->item( 0 ) );
```

Comment:
So it works like so:

- say `$node` has 2 child nodes.
- get the first child node 
- append to `$h2` removing it from `$node`
- `$node` now has 1 node. `h2` has 1 child node.
- Because number of child nodes is still more than 0, do it over.
- The second child of `$node` (now the first because we removed the actual first) gets added to end of h2
- once it has no more children, loop stops.
- not an infinite loop because we're moving the child node out of `$node`

This while loop syntax is just a bit neater than doing `foreach ( $node->childNodes )` as we don't have to do any checks to se whether there are any child nodes.